### PR TITLE
First pass at creating the model objects.

### PR DESCRIPTION
**Pull request by stefprez**
This PR also includes some attempted cleanup of necessary projects files and removing unnecessary project files. Hopefully we can get all this worked out soon so things like .classpath and .project file changes settle down. Hopefully I didn't muck anything up in the Checkstyle stuff in the process.

### CODE REVIEW

**Comment by csperlich** [comment]
This looks great, I say merge it. Will?

**Comment by Wubuntu88** [comment]
Great Job,  I just have a question about this code:
if (move instanceof Jump) {
                Jump jump = (Jump) move;

What is the relation between Move and Jump, why are you doing that type casting.  I admit I did not read all of the classes, I just didn't understand this part.

**Comment by stefprez** [comment]
So I created a class structure for the different types of moves that exist in Checkers. There is a MoveInterface which all types of moves implement. There is also a Jump interface that extends MoveInterface. The types of moves are: Move, SingleJump, and MultiJump. Move implements MoveInterface, while SingleJump and MultiJump implement Jump (which extends MoveInterface). The `instanceof` keyword in Java checks the actual type of the object at run time, since overloading method names with different classes for parameters doesn't work since Java doesn't do multiple dispatch. (I learned this the hard way.)

So basically I'm saying, "If this move is a jump, then lets cast it to a Jump object so I can call Jump specific methods on it."

**Comment by Wubuntu88** [comment]
Ok I get it.  Thanks for the description.  I didn't understand the last sentence of the first paragraph (the multiple dispatch thing), but I don't think that matters now.

Thanks, I merged it.  At some point, I'd like to figure out how I get the master code into my local repo, and how to get that on my computer.
